### ISSUE TITLE
YAMLLoadWarnin g: calling yaml.load() without Loader=... is deprecated

### ISSUE THREAD

**Issue by cschiri** [pymc-devs/pymc#3942]
## Description of problem:

Whenever I import pymc3, I get the following warning:

```
C:\Users\ixe\Anaconda3\lib\site-packages\dask\config.py:168: YAMLLoadWarnin
g: calling yaml.load() without Loader=... is deprecated, as the default Loader i
s unsafe. Please read https://msg.pyyaml.org/load for full details.
  data = yaml.load(f.read()) or {}
C:\Users\ixe\Anaconda3\lib\site-packages\distributed\config.py:20: YAMLLoad
Warning: calling yaml.load() without Loader=... is deprecated, as the default Lo
ader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  defaults = yaml.load(f)
```
Should I be worried about the warnings? How do I resolve the underlying issues; i.e. if they impede the functionality of pymc3?

## Versions and main components

* PyMC3 Version: 3.8
* Theano Version: 1.0.4
* Python Version: 3.7.3
* Operating system: Windows 8.1
* How did you install PyMC3: Pip Install

**Comment by ColCarroll** [comment]
This looks like it is a dask error, and I am confused as to why it would be issued when you import pymc3 (which does not require dask) -- did you share the full traceback, or are there more lines?

**Comment by cschiri** [comment]
Hi @ColCarroll 

It's not an error, it's a warning which is generated immediately when I import pymc3 in both the python console and the Jupyter notebook.  The warning is generated when I import the module without any other imports see:

![image](https://user-images.githubusercontent.com/31516469/83892087-71fe4b80-a74e-11ea-99b7-f5e34d0f94a8.png)

**Comment by rpgoldman** [comment]
@ColCarroll I suspect this is because pymc3 uses Arviz, which uses xarray, which uses dask. But that is just a guess

**Comment by ColCarroll** [comment]
dask is an optional xarray requirement, I think? @cschiri -- does the same think happen with just `import dask`? I am hopeful that this is actually their problem :D

**Comment by lucianopaz** [comment]
@cschiri, you should try to treat the warning as an error in order to the full traceback and identify which module is reasonable for it.

To do this you should run
```python
import warnings

with warnings.catch_warnings():
    warnings.simplefilter("error")
    import pymc3
```

Nevertheless, this must surely be a problem with xarray or dask itself. Maybe you are using an older version of xarray and a newer version of dask. Or simply an old version of dask and new version of pyyaml. Once you've tried handling the warning as an error, please post the full traceback you get.

**Comment by cschiri** [comment]
@lucianopaz 

I did as you suggested. Here is the full traceback:

```
---------------------------------------------------------------------------
RuntimeWarning                            Traceback (most recent call last)
<ipython-input-3-fa3197b2763d> in <module>
      3 with warnings.catch_warnings():
      4     warnings.simplefilter("error")
----> 5     import pymc3
      6 

~\Anaconda3\lib\site-packages\pymc3\__init__.py in <module>
      3 
      4 from .blocking import *
----> 5 from .distributions import *
      6 from .distributions import transforms
      7 from .glm import *

~\Anaconda3\lib\site-packages\pymc3\distributions\__init__.py in <module>
----> 1 from . import timeseries
      2 from . import transforms
      3 from . import shape_utils
      4 
      5 from .continuous import Uniform

~\Anaconda3\lib\site-packages\pymc3\distributions\timeseries.py in <module>
----> 1 from scipy import stats
      2 import theano.tensor as tt
      3 from theano import scan
      4 
      5 from pymc3.util import get_variable_name

~\Anaconda3\lib\site-packages\scipy\stats\__init__.py in <module>
    382 from __future__ import division, print_function, absolute_import
    383 
--> 384 from .stats import *
    385 from .distributions import *
    386 from .morestats import *

~\Anaconda3\lib\site-packages\scipy\stats\stats.py in <module>
    177 
    178 from scipy._lib.six import callable, string_types
--> 179 from scipy.spatial.distance import cdist
    180 from scipy.ndimage import measurements
    181 from scipy._lib._version import NumpyVersion

~\Anaconda3\lib\site-packages\scipy\spatial\__init__.py in <module>
     95 from __future__ import division, print_function, absolute_import
     96 
---> 97 from .kdtree import *
     98 from .ckdtree import *
     99 from .qhull import *

~\Anaconda3\lib\site-packages\scipy\spatial\kdtree.py in <module>
      6 import numpy as np
      7 from heapq import heappush, heappop
----> 8 import scipy.sparse
      9 
     10 __all__ = ['minkowski_distance_p', 'minkowski_distance',

~\Anaconda3\lib\site-packages\scipy\sparse\__init__.py in <module>
    230 from .csr import *
    231 from .csc import *
--> 232 from .lil import *
    233 from .dok import *
    234 from .coo import *

~\Anaconda3\lib\site-packages\scipy\sparse\lil.py in <module>
     18                       get_index_dtype, check_shape, check_reshape_kwargs,
     19                       asmatrix)
---> 20 from . import _csparsetools
     21 
     22 

__init__.pxd in init scipy.sparse._csparsetools()

RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 192 from C header, got 216 from PyObject
```

**Comment by cschiri** [comment]
@ColCarroll I did as you have suggested. Here is the output.

![image](https://user-images.githubusercontent.com/31516469/83925273-0b495400-a787-11ea-94a0-3a5c7b2dba8e.png)

I will try to upgrade dask, xarray and pyyaml and see if it makes a difference.

**Comment by cschiri** [comment]
How I fixed it.

Since I use Python through the Anaconda distribution, I updated pyyaml using the "conda update pyyaml" command. 

I also updated my versions of xarray and dask using the normal "pip install --upgrade [insert_module}" route. 

However, this created compatibility and stability challenges that were resolved by uninstalling dask from the environment.  It works, I made a trade-off, speed (via dask's parallel processing) for stability. Works like a charm now!

Thanks @ColCarroll and @lucianopaz for pointing me in the right direction.
